### DISCUSSION TITLE
Attachments not generated if tests runs in jar

### DISCUSSION THREAD

**Discussion by mage11** [allure-framework/allure2#1754] [Bug Reports] [answered]
Attachments generated if:
- run method/class in IDEA
- run "mvn test"

Attachments not generated if:
- run tests in jar

I use @Attachment, Allure.addAttachment()

How I run jar: java -jar mybuild.jar -testjar -xmlpathinjar testng,xml

No errors.

same problem:
https://stackoverflow.com/questions/66351148/attachments-annotations-step-screenshot-not-getting-saved-in-allure-reults

**Comment by mage11** [comment]
Hellooooooooo

**Comment by baev** [comment, marked as answer]
The `@Step` and `@Attachment` annotations are implemented using AspectJ. To ensure they function correctly when running the program from a jar file, you must define the AspectJ Weaver Java agent:

```bash
$ java -jar -javaagent:agentjweaver.jar tests.jar
```

Note: When running the program using Maven, the `argLine` configuration option of the `maven-surefire-plugin` is utilized for this purpose.

Answered in linked StackOverflow question as well: https://stackoverflow.com/a/77865736/3915166

**Reply by mage11** [reply]
thx
